subsys/usb/class/audio/audio.c

```diff
@@ -717,7 +717,8 @@ static int audio_custom_handler(struct usb_setup_packet *pSetup, int32_t *len,
 		 * must be the second interface associated with the device.
 		 */
 		if_desc = (struct usb_if_descriptor *)((uint8_t *)if_desc +
-						USB_ACTIVE_IF_DESC_SIZE);
+						USB_ACTIVE_IF_DESC_SIZE +
```

Comment:
ep_desc is only used to check if it is an input or output, it does so by checking if 0x80 is set in the endpoint address.  When using the wrong pointer instead bNrChannels from format_type_i_descriptor is checked and since there never is that many channels it will result in that the second channel is output, which is always true for bidirectional units.

Reply:
So this is definitely a bug. Could You please extract this one particular change into separate PR that actually fixes a bug?